Extensions/WikiUpdater/WikiUpdaterTask/src/GitWikiFuntions.ts

```diff
@@ -67,21 +67,36 @@ export function GetWorkingFile(filename, logInfo): any {
     return name;
 }
 
-export async function UpdateGitWikiFile(repo, localpath, user, password, name, email, filename, message, contents, logInfo, logError, replaceFile, appendToFile, tagRepo, tag) {
+export async function UpdateGitWikiFile(repo, localpath, user, password, name, email, filename, message, contents, logInfo, logError, replaceFile, appendToFile, tagRepo, tag, injectExtraHeader) {
     const git = simplegit();
 
     let remote = "";
-    let logremote = ""; // make sure we hide the password
-    if (password === null) {
+    let logremote = ""; // used to make sure we hide the password in logs
+    var extraHeaders = [];  // Add handling for #613
+
+    if (injectExtraHeader) {
         remote = `https://${repo}`;
         logremote = remote;
-    } else if (user === null) {
-        remote = `https://${password}@${repo}`;
-        logremote = `https://***@${repo}`;
+        if (user === null) {
+            extraHeaders = [`-c http.extraheader="AUTHORIZATION: bearer ${password}"`];
+            logInfo(`Adding the parameter '-c http.extraheader="AUTHORIZATION: bearer ***"' to clone operations`);
+        } else {
+            extraHeaders = [`-c http.extraheader="AUTHORIZATION: bearer ${user}:${password}"`];
```

Comment:
The bearer should only contain the access token which is ${password}

Reply:
OK I misunderstood, easy to change

Reply:
The user is useless in this case

Reply:
I have simplified the logic in line with the comments

Reply:
Yes this should work.

Reply:
Is it possible for me to test the plugin before an official release ?

Reply:
Yes if you are using on-premise Azure DevOps Server I can upload the VSIX or give me a Azure DevOps organisation name and I can share the private versions

Reply:
It is in fact an Azure Devops on-premise. Will you tell me when you upload the VSIX ? Thanks

Reply:
The Azure DevOps project that does the build is on a public repo. You can download the build artifact whihc has the public VSIX from https://dev.azure.com/richardfennell/GitHub/_build/results?buildId=4210&view=results

Reply:
Sorry that was the wrong link, give me a minute

Reply:
It will be the build https://dev.azure.com/richardfennell/GitHub/_build/results?buildId=4211&view=results when complete you can download the build artifacts these include the VSIX

Reply:
Unfortunately, it does not work even though it works when I run the command with a script task.
```
      - script: |
          git -c http.extraheader="AUTHORIZATION: bearer $(system.accesstoken)" clone $(myreo)
```
It is really frustrating, I don't understand

Reply:
I get this error:
```
fatal: unable to access 'repo_url': The requested URL returned error: 400
```

Reply:
Needs some more thought then it seems. It might be a limitation of the SimpleGit Library I am using. Need to read the documentation a bit more

Reply:
It is in fact the SimpleGit Library. It appears we have to use single quotes for the config value instead of double quotes. like so
```
extraHeaders = [`-c http.extraheader='AUTHORIZATION: bearer ${password}'`];
```